doc: add nfc rpc (#18155)
nrfconnect/sdk-nrf: doc/nrf/libraries/nfc/rpc/rpc.rst

.. _nfc_rpc:

NFC Remote Procedure Call
#########################

.. contents::
   :local:
   :depth: 2

> The :ref:`NFC <ug_nfc>` Remote Procedure Call (RPC) solution is a set of libraries that allows using the NFC running entirely on a separate device or CPU.

Author: I see already in some places using `NFC stack`. I guess it si possible to use.

You can use the following files as examples:

  * :file:`subsys/nfc/rpc/client/nfc_rpc_t2t_client.c`
  * :file:`subsys/nfc/rpc/server/nfc_rpc_t2t_server.c`

Requirements
************

These configuration options related to NFC Data Exchange Format must be enabled on the client for the Type 4 Tag:

  * :kconfig:option:`CONFIG_NFC_NDEF`
  * :kconfig:option:`CONFIG_NFC_NDEF_MSG`
  * :kconfig:option:`CONFIG_NFC_NDEF_RECORD`
  * :kconfig:option:`CONFIG_NFC_NDEF_TEXT_RECORD`

These configuration options related to NFC must be enabled on the server:

  * :kconfig:option:`CONFIG_NFC_T2T_NRFXLIB` - for Type 2 Tag
  * :kconfig:option:`CONFIG_NFC_T4T_NRFXLIB` - for Type 4 Tag

Samples using the library
*************************

The following |NCS| samples use this library:

* :ref:`nrf_rpc_protocols_serialization_client`
* :ref:`nrf_rpc_protocols_serialization_server`

Limitations
***********

The library currently supports serialization of the following:

  * :ref:`nrfxlib:type_2_tag`
  * :ref:`nrfxlib:type_4_tag`

The behavior of NFC with RPC is almost the same as without it but with the following exceptions:

>   * A couple of NFC API functions get data by pointer, for example :c:func:`nfc_t4t_ndef_rwpayload_set`. After calling the functions on the client, the data is sent to the server. Any manipulation with data over the pointer will not have any effect with server's data instance.

Reviewer: ```suggestion
  * ASome NFC API functions get data by pointer, for example :c:func:`nfc_t4t_ndef_rwpayload_set`.
    After calling the functions on the client, the data is sent to the server.
    Any manipulation of data over the pointer will not affect the server's data instance.
```

>   * Despite the maximum Tag 4 Type payload is possible to be 65520 bytes, the real length is limited by :c:macro:`NDEF_FILE_SIZE`.

Reviewer: ```suggestion
  * Despite the maximum payload for Type 4 Tag being 65520 bytes, the real length is limited by :c:macro:`NDEF_FILE_SIZE`.
```

Reviewer: To adapt Pekka's suggestion, this would be
```
  * Even though the maximum payload for Type 4 Tag can be 65520 bytes, the real length is limited by :c:macro:`NDEF_FILE_SIZE`.
